System.Server/Services/OrderService.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System.Server.Data;
+using System.Server.IServices;
+using System.Server.Models;
+using System.Server.Models.DTO;
+using System.Server.Enums;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
+
+namespace System.Server.Services
+{
+    public class OrderService : IOrderService
+    {
+
+        private readonly SystemContext _context;
+        public OrderService(SystemContext context)
+        { 
+            _context = context; 
+        }
+
+        public async Task<IEnumerable<OrderGetDTO>> GetAllOrders()
+        {
+            var orders = await _context.Orders.ToListAsync();
+            var orderList = new List<OrderGetDTO>();
+            var n = orders.Count;
+            for (int i = 0; i < n; i++)
+            {
+                orderList.Add(await GetOrderById(orders[i].Id));
+            }
+            return orderList;
+        }
+        public async Task<IEnumerable<OrderGetDTO>> GetAllUnpaidOrders()
+        {
+            var orders = await _context.Orders
+                .Where(o => o.Status == OrderStatus.Open)
+                .ToListAsync();
+            var orderList = new List<OrderGetDTO>();
+            var n = orders.Count;
+            for (int i = 0; i < n; i++)
+            {
+                orderList.Add(await GetOrderById(orders[i].Id));
+            }
+            return orderList;
+        }
+        public async Task<OrderGetDTO> GetOrderById(long id)
+        {
+            var order =  await _context.Orders.FindAsync(id);
+            if (order == null)
+            {
+                throw new KeyNotFoundException($"Order with ID {id} not found.");
+            }
+            var newOrder = new OrderGetDTO
+            {
+                Id = order.Id,
+                Status = order.Status,
+                Tip = order.Tip,
+                ReservationId = order.ReservationId
+            };
+
+            // Find order Products
+            var orderProducts = await _context.OrderProducts
+                .Where(p => p.OrderId == order.Id)
+                .ToListAsync();
+            var newProducts = new List<OrderGetProductDTO>();
+            var n = orderProducts.Count;
+            for (int i = 0; i < n; i++)
+            {
+                var product = await _context.Products
+                    .FindAsync(orderProducts[i].ProductId);
+                newProducts.Add(new OrderGetProductDTO
```

Comment:
man sioje vietoje kartais ismeta klaida, raso product is null nors mano supratimu neturetu. Man nesigauna replikuot pastoviai sita error. Siuo momentu pats nelabai pagaunu kame klaida. Paprašysiu, jog truputi daugiau patestuotum dar sita, o dar geriau, jeigu nenaudom kituose metoduose sito metodo.

Reply:
i guess galiu pacheckint ar product yra, bet jei orderiai kuriami su egzistuojančiais produktais ir tie produktai neištrinami tai neturėtų būt klaidos